src/utils.rs

```diff
@@ -26,3 +27,36 @@ impl<T> Deref for Lazy<T> {
         self.cell.get_or_init(self.init)
     }
 }
+
+trait UpcastFrom<T: ?Sized> {
+    fn any_from(value: &T) -> &Self;
+    fn any_mut_from(value: &mut T) -> &mut Self;
+}
+
+pub trait AsAny<U: ?Sized> {
```

Comment:
Is this intended to be public API?

Reply:
you can not require private `trait` on `pub trait`. But you can not actually do anything with it.

In the future I think this particular trait could be all pub.